src/go/MONIT/monit.go

```diff
@@ -410,29 +491,29 @@ func queryURL(rurl string, headers [][]string, verbose int) Record {
 }
 
 // helper function to run ES/InfluxDB or MONIT query
-func run(rurl, token string, dbid int, dbname, query, esapi string, idx, limit, verbose int) Record {
+func run(p Params) Record {
```

Comment:
In this function you are passing params by value, i.e. the params struct will be copied. If you'll re-use the same params over and over in different function calls or functions it will make unnecessary copies. Instead, you can change parameters of the function to accept a pointer, i.e. `func run(p *Params)` in this case you will create `Params` structure once and pass a pointer to it to the functions which will make program more efficient (in terms of heap/stack allocation). The beauty of Golang (with respect to C++) is that you can use the same syntax while referring attributes of the struct if you pass a struct itself (by value) or pointer to it, in both cases you will use `p.Query` for example